fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CollectJoinConstraint.java

```diff
@@ -47,24 +50,17 @@ public class CollectJoinConstraint implements RewriteRuleFactory {
     @Override
     public List<Rule> buildRules() {
         return ImmutableList.of(
-            logicalRelation().thenApply(ctx -> {
-                LeadingHint leading = (LeadingHint) ctx.cascadesContext
-                        .getStatementContext().getHintMap().get("Leading");
-                if (leading == null) {
-                    return ctx.root;
-                } else if (leading.isSyntaxError()) {
-                    return ctx.root;
-                }
-                return ctx.root;
-            }).toRule(RuleType.COLLECT_JOIN_CONSTRAINT),
-
             logicalJoin().thenApply(ctx -> {
-                LeadingHint leading = (LeadingHint) ctx.cascadesContext
-                        .getStatementContext().getHintMap().get("Leading");
-                if (leading == null) {
+                if (!ctx.cascadesContext.isLeadingJoin()) {
                     return ctx.root;
                 }
+                LeadingHint leading = (LeadingHint) ctx.cascadesContext
+                            .getHintMap().get("Leading");
                 LogicalJoin join = ctx.root;
+                if (join.getJoinType().isNullAwareLeftAntiJoin()) {
```

Comment:
Pls add this restriction into our document tracking.

Reply:
done